solr/core/src/java/org/apache/solr/search/facet/FacetParser.java

```diff
@@ -138,21 +139,30 @@ public Object parseFacetOrStat(String key, Object o) throws SyntaxError {
     return parseFacetOrStat(key, type, args);
   }
 
+  public interface ParseHandler {
+    Object doParse(FacetParser<?> parent, String key, Object args) throws SyntaxError;
```

Comment:
btw, is there an other exemplar of such a naming convention like `doSometh` ? why not just `parse`?

Reply:
No particular strong feelings on this, I tend to use `doSomething` when it's an inner helper method called from an outer/wrapper method that is the `something(..)` but in this case it's not really directly called. In practice I think people would use lambda anyway and never actually write this method name. But happy to change it to `parse`!